src/main/java/eu/dissco/backend/repository/SpecimenRepository.java

```diff
@@ -21,23 +23,57 @@ public class SpecimenRepository {
   private final DSLContext context;
   private final ObjectMapper mapper;
 
-  public List<DigitalSpecimen> getSpecimensLatest(int pageNumber, int pageSize) {
+  public List<JsonApiData> getSpecimensLatest(int pageNumber, int pageSize) {
     int offset = getOffset(pageNumber, pageSize);
     return context.select(NEW_DIGITAL_SPECIMEN.asterisk())
         .from(NEW_DIGITAL_SPECIMEN)
         .offset(offset)
         .limit(pageSize)
-        .fetch(this::mapper);
+        .fetch(this::mapToJsonApiData);
   }
 
-  public DigitalSpecimen getLatestSpecimenById(String id) {
+  public DigitalSpecimen getLatestSpecimenByIdObject(String id) {
     return context.select(NEW_DIGITAL_SPECIMEN.asterisk())
         .from(NEW_DIGITAL_SPECIMEN)
         .where(NEW_DIGITAL_SPECIMEN.ID.eq(id))
-        .fetchOne(this::mapper);
+        .fetchOne(this::mapToDigitalSpecimen);
   }
 
-  private DigitalSpecimen mapper(Record dbRecord) {
+  public JsonApiData getLatestSpecimenById(String id){
+    return context.select(NEW_DIGITAL_SPECIMEN.asterisk())
+        .from(NEW_DIGITAL_SPECIMEN)
+        .where(NEW_DIGITAL_SPECIMEN.ID.eq(id))
+        .fetchOne(this::mapToJsonApiData);
+  }
+
+  private JsonApiData mapToJsonApiData(Record dbRecord){
```

Comment:
I think it might be better to keep some structure here and first map it to a domain object, so we are sure all objects returned look the same. Then we can add that as the attributes in the JSON:API object